### ISSUE TITLE
Amplify push codegen doesn't preserve line separators

### ISSUE THREAD

**Issue by artyom-melnikov** [aws-amplify/amplify-cli#3069]
During `amplify push` following files (which are usually stored in git because they contain configuration) are overwritten:
* `amplify/backend/auth/<name>/template.json`
* `amplify/backend/api/<name>/parameters.json`

While the content is identical, git reports difference in line separators. 
On windows git with `autocrlf = true` option will convert LF to CRLF. Files generated by amplify use LF even on windows

Proposal: Use system line separators for generated files

**Comment by simon-lanf** [comment]
This sucks. It makes working with Windows and Linux really hard.

**Comment by kaustavghosh06** [comment]
@artyom-melnikov Is it due to some settings in your windows file editor?

**Comment by artyom-melnikov** [comment]
> @artyom-melnikov Is it due to some settings in your windows file editor?

I'm using WebStorm, it's not changing line endings in any way

**Comment by akshbhu** [comment]
@artyom-melnikov Can you check the default setting of Webstorm? is it LF or CRLF

**Comment by artyom-melnikov** [comment]
> @artyom-melnikov Can you check the default setting of Webstorm? is it LF or CRLF

For me it is CRLF (on Windows)

**Comment by JvdWest** [comment]
I struggled with the whole LF/CLRF issue with Amplify on Windows. I would do an Amplify pull and then a bunch of files would have "changes", but actually just LF/CRLF changes, and if you Git revert those guys, then you would have Amplify changes. I did change settings in my IDE, but it was resetting the whole time when I do Amplify pull/push. One thing that worked for me was to remove all Git changes (e.g. git stash). Then this:

`git config core.autocrlf input`

Then amplify pull again. Then there were no Git changes, and all my Amplify resources were up to date.
